service/src/main/java/bio/terra/drshub/services/TrackingService.java

```diff
@@ -0,0 +1,71 @@
+package bio.terra.drshub.services;
+
+import bio.terra.bard.api.BardApi;
+import bio.terra.bard.model.Event;
+import bio.terra.bard.model.EventProperties;
+import bio.terra.common.iam.BearerToken;
+import bio.terra.drshub.config.DrsHubConfig;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.map.PassiveExpiringMap;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class TrackingService {
+  static final String APP_ID = "drshub";
+
+  private final BardApiFactory bardApiFactory;
+  private final boolean trackInMixpanel;
+
+  private final Map<String, String> bearerTokenCache =
+      Collections.synchronizedMap(new PassiveExpiringMap<>(15, TimeUnit.MINUTES));
+
+  public TrackingService(BardApiFactory bardApiFactory, DrsHubConfig config) {
+    this.bardApiFactory = bardApiFactory;
+    this.trackInMixpanel = config.trackInMixPanel();
+  }
+
+  @Async("asyncExecutor")
```

Comment:
Is using the same executor as the requests going to cut into how performant DRSHub is?

Reply:
I was wondering about that.  I kinda doubt it since it's already set to Tomcat max threads.  One more reason to bump the JDK and us virtual threads